external/vcm/tests/test_coarsen_restarts.py

```diff
@@ -0,0 +1,142 @@
+import synth
+import xarray as xr
+
+from pathlib import Path
+from vcm.cubedsphere.coarsen_restarts import (
+    coarsen_restarts_on_sigma,
+    coarsen_restarts_on_pressure,
+    coarsen_restarts_via_blended_method,
+)
+from vcm.xarray_loaders import open_json, to_json
+
+DIR_NAME = os.path.dirname(__file__)
+SCHEMA_PATH = os.path.join(DIR_NAME, "_coarsen_restarts_regression_tests/schemas")
+REFERENCE_PATH = os.path.join(DIR_NAME, "_coarsen_restarts_regression_tests/reference")
+
+FACTOR = 2
+RANGES = {
+    "delp": synth.Range(3, 5),
+    "area": synth.Range(0.5, 1),
+    "dx": synth.Range(0.5, 1),
+    "dy": synth.Range(0.5, 1),
+}
+RESTART_CATEGORIES = ["fv_core.res", "fv_tracer.res", "fv_srf_wnd.res", "sfc_data"]
+REGRESSION_TESTS = {
+    "mass-weighted-model-level-with-agrid-winds": (
+        coarsen_restarts_on_sigma,
+        {"coarsen_agrid_winds": True, "mass_weighted": True},
+    ),
+    "area-weighted-model-level-without-agrid-winds": (
+        coarsen_restarts_on_sigma,
+        {"coarsen_agrid_winds": False, "mass_weighted": False},
+    ),
+    "pressure-level-with-agrid-winds": (
+        coarsen_restarts_on_pressure,
+        {"coarsen_agrid_winds": True},
+    ),
+    "pressure-level-without-agrid-winds": (
+        coarsen_restarts_on_pressure,
+        {"coarsen_agrid_winds": False},
+    ),
+    "blended-mass-weighted-with-agrid-winds": (
+        coarsen_restarts_via_blended_method,
+        {"coarsen_agrid_winds": True},
+    ),
+    "blended-area-weighted-without-agrid-winds": (
+        coarsen_restarts_via_blended_method,
+        {"coarsen_agrid_winds": False, "mass_weighted": False},
+    ),
+}
+
+
+def open_restart_schemas(schema_path):
+    schemas = {}
+    for category in RESTART_CATEGORIES:
+        filename = os.path.join(schema_path, f"{category}-schema.json")
+        with open(filename, "r") as file:
+            schemas[category] = synth.load(file)
+    return schemas
+
+
+def open_grid_spec_schema(schema_path):
+    filename = os.path.join(schema_path, "grid-schema.json")
+    with open(filename, "r") as file:
+        return synth.load(file)
+
+
+def generate_synthetic_restart_data(schemas):
+    data = {}
+    for category, schema in schemas.items():
+        data[category] = synth.generate(schema, RANGES).compute()
+    return data
+
+
+def generate_synthetic_grid_spec_data(schema):
+    return synth.generate(schema, RANGES).compute()
+
+
+def reference_json(root, tag, category):
+    return os.path.join(root, f"{tag}-{category}.json")
+
+
+def open_reference_data(root, tag):
+    data = {}
+    for category in RESTART_CATEGORIES:
+        filename = reference_json(root, tag, category)
+        data[category] = open_json(filename)
+    return data
+
+
+def generate_synthetic_data(restart_schemas, grid_schema):
+    restart_data = generate_synthetic_restart_data(restart_schemas)
+    grid_data = generate_synthetic_grid_spec_data(grid_schema)
+    return restart_data, grid_data
+
+
+@pytest.mark.slow
```

Comment:
These tests are now less than 1.5 seconds each in CI (and less than 1 second each on my VM).  I don't think it does much harm to keep it, but we could remove this mark if we considered it to be a bit of a distraction now.

Reply:
Yeah, I'd say leave it.